src/matlab2tikz.m

```diff
@@ -4082,16 +4105,27 @@ function warnFacetedInterp(m2t)
     % Toggle legend entry
     drawOptions = maybeShowInLegend(m2t.currentHandleHasLegend, drawOptions);
 
+    drawOpts = opts_print(drawOptions);
+    
     % Generate the tikz table
     xData = get(h, 'XData');
     yData = get(h, 'YData');
-    [m2t, table, tableOptions] = makeTable(m2t, '', xData, '', yData);
+    if m2t.axes{end}.is3D
+        % TODO: account for hgtransform
+        zData = get(h, 'ZData');
+        [m2t, table, tableOptions] = makeTable(m2t, '', xData, '', yData, '', zData);
+        % Print out
+        tabOpts  = opts_print(tableOptions);
+        str = sprintf('\\addplot3 [%s]\n table[%s] {%s};\n ', ...
+                         drawOpts, tabOpts, table);
+    else
+        [m2t, table, tableOptions] = makeTable(m2t, '', xData, '', yData);
+        % Print out
+        tabOpts  = opts_print(tableOptions);
+        str = sprintf('\\addplot[%s] plot table[%s] {%s};\n', ...
```

Comment:
I will tackle those in a separate PR: then it's just a question of search-and-replace-and-update-hashes.